### PR TITLE
Updated docs for server infrastructure and backups

### LINKED ISSUE #2364: CI/CD: document the design
Add or update documentation about CI/CD pipelines and their setup to AMY docs (https://carpentries.github.io/amy/)

### LINKED ISSUE #2365: Document database snapshot and backup settings
Add production database server snapshot and backup settings in AMY documentation (https://carpentries.github.io/amy/). Consider some options for better handling of finer-grain backups.

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
This fixes #2364.
This fixes #2365.

### CODE REVIEW

**Comment by maneesha** [comment]
I will defer to @elichad  on this to be sure everything we need to know is covered.

File: docs/design/server_infrastructure.md [hunk 1]
```diff
@@ -73,25 +71,64 @@ that with:
...
+
+### AWS services used
+
+* [ECS](https://aws.amazon.com/ecs/) - for running a container with AMY with a load
+  balancer
+* [RDS](https://aws.amazon.com/rds/) - for database (serverless PostgreSQL-flavored Aurora)
+* [CloudFront](https://aws.amazon.com/cloudfront/) is used as CDN for assets
+* [Route 53](https://aws.amazon.com/route53/) for keeping domain records and managing
+  TLS certificates
+* [ECR](https://aws.amazon.com/ecr/) - for storing Docker images
```
Comment (elichad) [inline docs/design/server_infrastructure.md:107]: Could you include what these acronyms stand for? It's tough to keep track of them all!

**Review by elichad — COMMENTED** [review]
This is a helpful writeup, thanks! Couple of small comments, otherwise I'm happy with this.

File: docs/design/server_infrastructure.md (hunk 1 above)
Comment (pbanaszkiewicz) [reply docs/design/server_infrastructure.md]: Good point, updated.

File: docs/design/database_backups.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+A good practice is to create a manual snapshot in AWS console before starting
+a deployment. Manual snapshots are not deleted automatically.
+
+## Automatic backup from inside of the server
+
+A script is set up to run every 30 minutes and upload a `pg_dump` of the database to S3.
+
+> **Note March 2023:** this script seems to be broken. It is a good idea to revise if
+> it's still needed.
```
[the remark this answers is not in the document]
Comment (pbanaszkiewicz) [reply docs/design/database_backups.md]: Thanks, I removed that note.

**Comment by pbanaszkiewicz** [comment]
@elichad It's ready for your re-review, thanks.

**Review by elichad — APPROVED** [review]

File: docs/design/server_infrastructure.md (hunk 1 above)
Comment (elichad) [reply docs/design/server_infrastructure.md]: Much better, thanks!

**Pull request merged**